public/app/features/panel/panellinks/specs/link_srv.test.ts

```diff
@@ -137,4 +138,38 @@ describe('linkSrv', () => {
       ).toEqual('/d/1?time=1000000001');
     });
   });
+
+  describe('sanitization', () => {
+    const url = "javascript:alert('broken!);";
+    it.each`
+      disableSanitizeHtml | expected
+      ${true}             | ${url}
+      ${false}            | ${'about:blank'}
+    `(
+      "when disable disableSanitizeHtml set to '$disableSanitizeHtml' then result should be '$expected'",
+      ({ disableSanitizeHtml, expected }) => {
+        const restoreConfig = mockConfig({
+          disableSanitizeHtml,
+        });
+
+        const link = linkSrv.getDataLinkUIModel(
+          {
+            title: 'Any title',
+            url,
+          },
+          {
+            __value: {
+              value: { time: dataPointMock.datapoint[0] },
+              text: 'Value',
+            },
+          },
+          {}
+        ).href;
+
+        // console.log(link);
+        expect(link).toBe(expected);
+        restoreConfig();
```

Comment:
are not all jest test run in some form of isolation? do we need to the restoreConfig here?

Reply:
Good point. In this case it is not necessary. But if other parts of `link_srv` were dependant on `config` and some other property than `disableSanitizeHtml` than not restoring the partial config might have caused the tests to fail. So I consider this more of a good practice rather than necessity. WDYT?

Reply:
having to restore state after asserts looks to me like a smell / bad test design, I would add a common setup that set's the expected state. or have a common setup / scenario function that let you pass in expected config.